src/types.ts

```diff
@@ -307,68 +307,118 @@ export enum ANT_INTERACTION_TYPES {
 }
 
 export type ContractType = (typeof CONTRACT_TYPES)[keyof typeof CONTRACT_TYPES];
+
 export type AntInteraction =
   (typeof ANT_INTERACTION_TYPES)[keyof typeof ANT_INTERACTION_TYPES];
 export type RegistryInteraction =
   (typeof REGISTRY_INTERACTION_TYPES)[keyof typeof REGISTRY_INTERACTION_TYPES];
 
-export type TransactionData =
-  | ({
-      assetId: string;
-      function: string;
-    } & BuyRecordPayload)
-  | ExtendLeasePayload
-  | UpgradeTierPayload
-  | TransferIOPayload
-  | SetTickerPayload
-  | SetControllerPayload
-  | SetNamePayload
-  | SetRecordPayload
-  | RemoveRecordPayload
-  | TransferAntPayload
-  | CreateAntPayload;
+export const ALL_TRANSACTION_DATA_KEYS = [
+  'assetId',
+  'functionName',
+  'deployedTransactionId',
+  'name',
+  'contractTxId',
+  'years',
+  'tierNumber',
+  'target',
+  'qty',
+  'ticker',
+  'subDomain',
+  'transactionId',
+  'ttlSeconds',
+  'srcCodeTransactionId',
+  'initialState',
+  'tags',
+];
+
+export type TransactionData = {
+  assetId: string;
+  functionName: string;
+  deployedTransactionId?: ArweaveTransactionID;
+} & BuyRecordPayload &
+  ExtendLeasePayload &
+  UpgradeTierPayload &
+  TransferIOPayload &
+  SetTickerPayload &
+  SetControllerPayload &
+  SetNamePayload &
+  SetRecordPayload &
+  RemoveRecordPayload &
+  TransferAntPayload &
+  CreateAntPayload;
 
 // TODO: discuss bellow setup with dylan
 //export const TRANSACTION_DATA_KEYS = new Map<ContractType, AntInteraction | RegistryInteraction>()
 
 export const TRANSACTION_DATA_KEYS: {
   // specifying interaction types to the correct contract type, to ensure clarity and to prevent crossover of interaction types
-  [CONTRACT_TYPES.ANT]: { [K in AntInteraction]: string[] };
-  [CONTRACT_TYPES.REGISTRY]: { [K in RegistryInteraction]: string[] };
+  [CONTRACT_TYPES.ANT]: {
+    [K in AntInteraction]: { functionName: string; keys: string[] };
+  };
+  [CONTRACT_TYPES.REGISTRY]: {
+    [K in RegistryInteraction]: { functionName: string; keys: string[] };
```

Comment:
nit: `keys` might be better expressed as `requiredKeys` - also, these are duplicate types, so can abstract into one type

Reply:
I would agree, but not all of these keys are actually required per contract code - adding the type though

Reply:
i mean:

```
export type InteractionInput = {
 functionName: string;
 keys: string[]
```
and using it as a type for lines 357 and 360